zulu/datetime.py

```diff
@@ -567,6 +568,114 @@ def span(self, frame, count=1):
         """
         return (self.start_of(frame), self.end_of(frame, count))
 
+    @classmethod
+    def validate_times(cls, start, end):
+        """Class method that helps in validating the given start and end
+        datetimes.
+        """
+        if not start or not isinstance(start, datetime):
+            raise ValueError('Provided value {0} is invalid datetime'
+                             .format(start))
+
+        if not end or not isinstance(end, datetime):
+            raise ValueError('Provided value {0} is invalid datetime'
+                             .format(end))
+
+        if start > end:
+            raise ValueError('Start datetime should always be less than end '
+                             'datetime')
+
+    @classmethod
+    def span_range(cls, frame, start=None, end=None):
+        """Class method that helps in getting a range of time spans.
+
+        Args:
+            frame (str): A time frame. Ex: year, month, day, minute, etc.
+            start (datetime): A start datetime value.
+            end (datetime): An end datetime value.
+
+        Returns:
+            list: List of all time spans
+        """
+        # Validate the start datetime, end datetime and the given frame.
+        cls.validate_times(start, end)
+        cls.validate_frame(frame)
+
+        span_list = []
+        is_less_than_end_time = True
+
+        # This loop helps in bringing all the time spans with the given time
+        # frame.
+        while is_less_than_end_time:
+            # We use the span() method to bring the actual tuple with the
+            # given time frame.
+            time_span_tuple = cls.fromdatetime(start).span(frame)
+
+            # We change the value of start date to the floor value of the
+            # returned tuple from span() method.
+            start = cls.fromdatetime(time_span_tuple[1])
```

Comment:
I would probably alias the `start` variable to `next` (here and before the loop) so it's a little clearer that it's not the same thing as the input variable `start`; it's the "next" span in the range.